controller/tasks.go

```diff
@@ -241,6 +245,40 @@ func (c *Controller) getTaskHandler(w http.ResponseWriter, r *http.Request) {
 	dagjson.Encoder(task.Representation(), w)
 }
 
+func (c *Controller) carHandler(w http.ResponseWriter, r *http.Request) {
+	logger := log.With("req_id", r.Header.Get("X-Request-ID"))
+
+	logger.Debugw("handle request", "command", "car")
+	defer logger.Debugw("request handled", "command", "car")
+
+	w.Header().Set("Content-Type", "application/octet-stream")
+
+	store := c.db.Store(r.Context())
+	rootCid, err := store.Head()
+	if err != nil {
+		log.Errorw("get task DB error", "err", err.Error())
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	w.WriteHeader(http.StatusOK)
+
+	ssb := builder.NewSelectorSpecBuilder(basicnode.Prototype__Any{})
+	ss := ssb.ExploreRecursive(selector.RecursionLimitNone(), ssb.ExploreUnion(
+		ssb.Matcher(),
+		ssb.ExploreAll(ssb.ExploreRecursiveEdge()),
+	))
+	root := car.Dag{
+		Root:     rootCid,
+		Selector: ss.Node(),
+	}
+	sc := car.NewSelectiveCar(r.Context(), store, []car.Dag{root})
```

Comment:
side note: we should make a version of selective car that actually just takes an IPLD loader -- under the hood, this is actually converting the reader BACK to a loader, which is funny case you converted a loader to a reader to get this to work :)